float32 left
float32 right
---
bool success
